festim/generic_simulation.py

```diff
@@ -273,6 +274,20 @@ def initialise(self):
             self.mesh.dx, self.mesh.ds, self.materials
         )
 
+        # needed to ensure that data is actually exported at TXTExport.times
+        # see issue 675
+        for export in self.exports.exports:
+            if isinstance(export, festim.TXTExport) and export.times:
+                if not self.dt.milestones:
+                    self.dt.milestones = []
+                for time in export.times:
+                    if time not in self.dt.milestones:
+                        msg = "To ensure that TXTExport exports data at the desired times "
+                        msg += "TXTExport.times are added to milestones"
+                        warnings.warn(msg)
+                        self.dt.milestones.append(time)
+                self.dt.milestones.sort()
```

Comment:
As per my comment below, we may need a test that explicitly checks these lines. In other words, checking that `TXTExport` integrates well with `Simulation`

Reply:
Done